zerver/actions/message_send.py

```diff
@@ -1557,6 +1618,10 @@ def check_message(
 
         check_private_message_policy(realm, sender, user_profiles)
 
+        recipients_for_user_creation_events = get_recipients_for_user_creation_events(
+            realm, sender, user_profiles
```

Comment:
We compute this early here because huddle objects are created before sending a message and thus we would not get the correct output of accessible users after creation the huddle objects. But this means we have to pass this data to a couple of functions below in the stack.

Reply:
OK; let's put a version of this note into a comment just before the call. (In general, I think when posting a GitHub comment like this, it's always worth considering whether it makes sense to copy it into the code.